src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -35,8 +35,10 @@ public class RoslynAnalyzerProvider(
     public const string CSharpLanguage = "cs";
     public const string VBNetLanguage = "vbnet";
 
-    private const string LegacyServerPropertyPrefix = "sonaranalyzer-";
-    private const string RoslynRepoPrefix = "roslyn.";
+    private const string ServerPropertyFormat = "sonar.{0}.analyzer";
+    private const string RoslynRepoPrefix = "roslyn."; // Used for plugins generated by SonarQube Roslyn SDK, and legacy Sonar Security C# Frontend
+    private const string LegacySecurityFrontEndPropertyPrefix = "sonaranalyzer.security.cs"; // Should not be used and exists for backward compatibility with version <= 9.2
+    private const string LegacyServerPropertyPrefix = "sonaranalyzer-"; // Should not be used and exists for backward compatibility with version <= 9.2.
```

Comment:
What is version 9.2? Plugin? SonarQube? Scanner?
Plugin was < 10.4
SQ was < 10.8
Scanner is irrelevant
```suggestion
    private const string LegacyServerPropertyPrefix = "sonaranalyzer-"; // Backward compatibility with ??? version <= 9.2.
```